tests/integration/test_docker.py

```diff
@@ -220,3 +220,30 @@ def test_nonblocking_execute():
         gen = cont.execute(["printf", "asd\nasd"], blocking=False)
         assert [b"asd\nasd"] == list(gen)
         cont.execute(["bash", "-c", "sleep 0.01; exit 110"], blocking=False)
+
+
+def test_pull_always():
+    with DockerBackend() as backend:
+        # this is an old version, likely not present anywhere
+        image = backend.ImageClass("docker.io/library/busybox", tag="1.25.1",
+                                   pull_policy=DockerImagePullPolicy.ALWAYS)
+        try:
+            assert image.is_present()
+        finally:
+            image.rmi(force=True)
+
+
+def test_pull_if_not_present():
+    with DockerBackend() as backend:
+        image = backend.ImageClass("docker.io/library/busybox", tag="1.25.1")
+        try:
+            assert image.is_present()
+        finally:
+            image.rmi(force=True)
+
+
+def test_pull_never():
+    with DockerBackend() as backend:
```

Comment:
please add `assert not image.is_present()` also before the call so when this fails we know that it is cause of functionality of image __init__, not presence of busybox on localhost.